swift/ql/src/queries/Security/CWE-943/PredicateInjection.qhelp

```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE qhelp PUBLIC "-//Semmle//qhelp//EN" "qhelp.dtd">
+<qhelp>
+<overview>
+<p>
+Predicates represent logical conditions that can be used to check whether an object matches them.
+If a predicate is built from user-provided data without sufficient sanitization, an attacker may
+be able to change the overall meaning of the predicate.
+</p>
+</overview>
+<recommendation>
+<p>
+When building a predicate from untrusted data, pass it to the appropriate <code>arguments</code> parameter
+during initialization, or as an array of substitution variables before evaluation, instead of appending
+or concatenating it to the body of the predicate.
+</p>
+</recommendation>
+<example>
+<p>
+In the following insecure example, a <code>NSPredicate</code> is built directly from data obtained from an HTTP request,
+which is untrusted, and can be arbitrarily set by an attacker to alter the meaning of the predicate:
+</p>
+<sample src="PredicateInjectionBad.swift" />
+<p>
+A better way to do this is using the <code>arguments</code> parameter of <code>NSPredicate</code>'s constructor,
+which prevents attackers from altering the meaning of the predicate, even if they control the externally-obtained data,
+as seen in the following secure example:
+</p>
+<sample src="PredicateInjectionGood.swift" />
+</example>
+<references>
+<li>Apple Developer documentation: <a href="https://developer.apple.com/documentation/foundation/nspredicate">NSPredicate</a> </li>
```

Comment:
```suggestion
<li>Apple Developer Documentation: <a href="https://developer.apple.com/documentation/foundation/nspredicate">NSPredicate</a> </li>
```

Minor: I guess if we're capitalizing `Developer` we should capitalize `Documentation`, too, as Apple themselves seem to.